src/hotspot/share/opto/type.hpp

```diff
@@ -181,9 +181,12 @@ class Type {
   const Type *hashcons();       // Hash-cons the type
   virtual const Type *filter_helper(const Type *kills, bool include_speculative) const;
   const Type *join_helper(const Type *t, bool include_speculative) const {
+    assert_type_verify_empty();
     return dual()->meet_helper(t->dual(), include_speculative)->dual();
   }
 
+  void assert_type_verify_empty() const PRODUCT_RETURN;
```

Comment:
Just noticed that `PRODUCT_RETURN` breaks optimized build (`Type::assert_type_verify_empty()` is under `ASSERT`).